src/SessionSetup/RetrieveInstancesWidgetDemo/main.cpp

```diff
@@ -33,10 +33,9 @@ int main(int argc, char* argv[]) {
   std::shared_ptr<orbit_qt_utils::MainThreadExecutorImpl> executor{
       orbit_qt_utils::MainThreadExecutorImpl::Create()};
 
-  std::unique_ptr<RetrieveInstances> retrieve_instances =
-      RetrieveInstances::Create(client_ptr, executor.get());
+  RetrieveInstancesWidget widget{};
 
-  RetrieveInstancesWidget widget{retrieve_instances.get()};
+  widget.SetRetrieveInstances(RetrieveInstances::Create(client_ptr, executor.get()));
```

Comment:
We will always SetRetrieveInstances immediately after creating the widget?

Reply:
Seems strange to me, but I don't have a big knowledge of QDesigner, could be useful to have a better review here. Meanwhile, since you look at it and seems sure I won't concern about it.